extension/storage/filestorage/config.go

```diff
@@ -65,7 +65,7 @@ func (cfg *Config) Validate() error {
 	for _, dir := range dirs {
 		if info, err := os.Stat(dir); err != nil {
 			if !cfg.CreateDirectory && os.IsNotExist(err) {
-				return fmt.Errorf("directory must exist: %w", err)
+				return fmt.Errorf("directory must exist: %w. You can enable the create_directory option to automatically create it.", err)
```

Comment:
Actually this whole logic is used not only for the `directory` configuration option, but also for the `compaction::directory` option. Only the validation of the `directory` option should be modified. The validation for `compaction::directory` setting should be unchanged.

Reply:
@andrzej-stencel we also create the compaction directory, if it doesn't exist. See here https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/49f80150927b02c2ab2c78bd655a42db14767ee8/extension/storage/filestorage/extension.go#L30-L47

---
On a side note, I've updated docs to mention the same
> `create_directory` when set, will create the data storage and compaction directory if it does not already exist. The directory will be created with `0750 (rwxr-x--)` permissions, by default. Use `directory_permissions` to customize directory creation permissions.

Reply:
Oh, missed that, thanks. I suppose that's OK, thanks for updating the docs.

I see the following issue in this logic though. The compaction directory is only created when `compaction::on_start` is enabled, but not when `compaction::on_rebound` is enabled. This means that if I use the following configuration:

```yaml
extensions:
  file_storage:
    compaction:
      directory: non-existent-compaction-dir
      on_rebound: true
    create_directory: true
    directory: non-existent-storage-dir 
```

then the collector will not create the compaction directory `non-existent-compaction-dir` either on start or later, when rebound compaction is being performed, resulting in an error trying to compact.

I think ideally in this case the compaction directory should be created not on start, but only if/when needed, if/when the rebound compaction is actually performed.

Alternatively, only have the `create_directory` option affect the root `directory` option, leaving space for creating a separate `compaction::create_directory` option controlling the creation of the compaction directory. I would be fine with either approach.

Reply:
> I see the following issue in this logic though. The compaction directory is only created when compaction::on_start is enabled, but not when compaction::on_rebound is enabled.

@andrzej-stencel I filed https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35114 for this.

As per my understanding, we should check for compaction directory when either of the options (`on_start` or `on_rebound`) are enabled

Reply:
with https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35114 merged, we now create compaction on rebound or start.

> I think ideally in this case the compaction directory should be created not on start, but only if/when needed, if/when the rebound compaction is actually performed.

Rebound compaction takes place periodically, usually every 5 seconds by default. Checking the compaction directory with every iteration would be an inefficient. What do you think?

Reply:
> Rebound compaction takes place periodically, usually every 5 seconds by default.

This is inaccurate. The check takes place periodically, but the compaction is only triggered when the conditions are met.

But I suppose creating the directory on collector's start when rebound compaction is enabled is OK too.

Thank you for making these changes and for the tests Vihas! :pray:

Reply:
> This is inaccurate. The check takes place periodically, but the compaction is only triggered when the conditions are met.

Yes, sorry I missed that.